### PR TITLE
grv 0.1.2 (new formula)

### PR DESCRIPTION
**Pull request by rgburke**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----
GRV is a terminal interface for viewing git repositories

### CODE REVIEW

File: Formula/grv.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
+class Grv < Formula
+  desc "Terminal interface for viewing git repositories"
+  homepage "https://github.com/rgburke/grv"
+  url "https://github.com/rgburke/grv/releases/download/v0.1.2/grv-0.1.2-src.tar.gz"
+  sha256 "1f2f8adb28085fd892f4b23cf5cf17925fc502cf479be35193e1c6b93cad1e49"
+  head "https://github.com/rgburke/grv.git"
+
+  depends_on "go" => :build
+  depends_on "pkg-config" => :build
+  depends_on "cmake" => :build
```
Comment (fxcoudert) [inline Formula/grv.rb]: alphabetical order among build deps

File: Formula/grv.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    path.install buildpath.children
+
+    cd path do
+      system "make", "build-only"
+      bin.install "grv"
+    end
+  end
+
+  test do
+    assert_match version.to_s, shell_output("#{bin}/grv -version")
```
Comment (fxcoudert) [inline Formula/grv.rb]: We need a test that does a bit more than output version number.

**Review by fxcoudert — CHANGES_REQUESTED** [review]

**Comment by rgburke** [comment]
@fxcoudert Thank you for the review and the feedback. I have corrected the order of the build deps and improved the test.

File: Formula/grv.rb [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  depends_on "readline"
+
+  def install
+    ENV["GOPATH"] = buildpath
+    path = buildpath/"src/github.com/rgburke/grv"
+    path.install buildpath.children
+
+    cd path do
+      system "make", "build-only"
+      bin.install "grv"
```
Comment (DomT4) [inline Formula/grv.rb]: You'll likely need to add a `prefix.install_metafiles` line here.

This ensures things like `README`, `LICENSE` and `CHANGELOG` where available are installed into the prefix. Normally Homebrew handles this automatically with a little magic, but when you relocate the entire contents of the `buildpath` to another directory for the install process as is common with `go`-using formulae that logic falls on its face & it needs a little manual nudging. It's a known issue.

File: Formula/grv.rb [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    end
+  end
+
+  test do
+    ENV["TERM"] = "xterm"
+
+    system "git", "init"
+    system "git", "commit", "--allow-empty", "-m", "test"
+    system "echo '<grv-exit>' | #{bin}/grv -logLevel DEBUG"
```
Comment (DomT4) [inline Formula/grv.rb]: Use `pipe_output()` for this.

**Review by fxcoudert — APPROVED** [review]

**Comment by fxcoudert** [comment]
Thanks @rgburke for the pull request, and congrats on your first contribution to Homebrew!

**Pull request closed without merging**